episodes/preparing_for_machine_learning.md

```diff
@@ -0,0 +1,438 @@
+from matplotlib import pyplot as plt
+from skimage import data
+from skimage import transform
+from skimage import io
+from skimage import img_as_float
+from skimage.transform import rotate
+from skimage import transform as tf
+from skimage.transform import PiecewiseAffineTransform
+```
+
+Then, we import our example images and examine them.
+
+```python
+image_b = io.imread('cardiomegaly_cc0.png') # cardiomegaly CXR
+image_g = io.imread('rotatechest.png') # a relatively normal CXR
+image_y = io.imread('other_op.png') # a relatively normal CXR
+# create figure
+fig = plt.figure(figsize=(10, 7))
+  
+# setting values to rows and column variables
+rows = 1
+columns = 3
+
+# Adds a subplot at the 1st position
+fig.add_subplot(rows, columns, 1)
+# showing image
+plt.imshow(image_g)
+plt.axis('off')
+plt.title("Normal 1")
+  
+# add a subplot at the 2nd position
+fig.add_subplot(rows, columns, 2)
+# showing image
+plt.imshow(image_b)
+plt.axis('off')
+plt.title("Cardiomegaly")
+
+# add a subplot at the 3nd position
+fig.add_subplot(rows, columns, 3)
+# showing image
+plt.imshow(image_y)
+plt.axis('off')
+plt.title("Normal 2")
+```
+
+![](fig/cxr_display_mip.png){alt='CXR examples'}
+
+
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Thought Challenge: Can you see some problems in the following scenario?
+
+Imagine you got the above images and many more because you have been assigned to make an algorithm for cardiomegaly detection so patients can be notified if by chance their X-rays acquired by the hospital for any purpose shows cardiomegaly. (This would be an example of using machine learning for opportunistic screening.) You are given one dataset of healthy (no cardiomegaly) patients who went to an outpatient clinic in a very poor area staffed with first year radiography students. Those patients were being checked for tuberculosis (TB). You also got chest X-rays of cardiomegaly patients from an extremely prestigeous tertiary inpatient hospital. 
+
+:::::::::::::::::::::::: solution 
+
+## Solution
+ 
+All of the following may be potential problems:
+
+- We may have different quality images from different machines for our healthy versus diseased in this scenario, this could be called a "batch effect" creating bias
+- In the presigous hospital many X-rays may be supine, will these combine well with standing APs?
+- We could suspect labelling may not be entirely accurate in the clinic because has by lower level staff 
+
+:::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Code Challenge: Using skimage.transform.rotate
+
+Use `skimage.tranform.rotate` to produce two realistic augmented images (name them `new_pic1` and `new_pic2`) from the given 'normal' image we put in `image_g` variable, and then apply what you percieve as the two most critical algorithms to make them ready for classic supervised machine learning in one bit of code. 
```

Comment:
Regarding applying the two most critical algorithms to make the images ready for classic ML, what algorithms are you referring to?
In the solution, I see only an image crop applied.

Reply:
This was an oops, I even forgot to rotate things... I have fixed it now.